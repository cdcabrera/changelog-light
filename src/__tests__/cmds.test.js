const { join } = require('path');
const cmds = require('../cmds');
const { OPTIONS } = require('../global');

describe('Commands', () => {
  let mockObjectClear;

  beforeEach(() => {
    const { mockClear } = mockObjectProperty(OPTIONS, {
      date: '2022-10-01',
      changelogPath: join(OPTIONS.contextPath, 'CHANGELOG.md'),
      packagePath: join(OPTIONS.contextPath, 'package.json'),
      releaseBranch: 'HEAD'
    });

    mockObjectClear = mockClear;
  });

  afterEach(() => {
    mockObjectClear();
  });

  it('should attempt to run commands', () => {
    const { mockClear } = mockObjectProperty(OPTIONS, {
      releaseTypeScope: 'chore(release)'
    });

    expect(
      Object.entries(cmds).map(([key, func]) => ({
        [key]: func()
      }))
    ).toMatchSnapshot('commands');

    mockClear();
  });

  it('should handle multiple formats for releaseTypeScope', () => {
    // string
    const { mockClear: strMockClear } = mockObjectProperty(OPTIONS, {
      releaseTypeScope: 'chore(release)'
    });

    expect({ commitFiles: cmds.commitFiles(), getReleaseCommit: cmds.getReleaseCommit() }).toMatchSnapshot(
      'releaseTypeScope, string'
    );

    strMockClear();

    // single list item
    const { mockClear: singleItemMockClear } = mockObjectProperty(OPTIONS, {
      releaseTypeScope: ['chore(release)']
    });

    expect({ commitFiles: cmds.commitFiles(), getReleaseCommit: cmds.getReleaseCommit() }).toMatchSnapshot(
      'releaseTypeScope, single list item'
    );

    singleItemMockClear();

    // multi-list item
    const { mockClear: multiItemMockClear } = mockObjectProperty(OPTIONS, {
      releaseTypeScope: ['chore(release)', 'dolor sit', 'lorem ipsum']
    });

    expect({ commitFiles: cmds.commitFiles(), getReleaseCommit: cmds.getReleaseCommit() }).toMatchSnapshot(
      'releaseTypeScope, multi-list item'
    );

    multiItemMockClear();
  });
});
