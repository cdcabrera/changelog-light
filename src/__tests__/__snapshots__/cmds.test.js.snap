// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Commands should attempt to run commands: commands 1`] = `
[
  {
    "commitFiles": "<execSync>["git add ../src/__fixtures__/package.json ../src/__fixtures__/CHANGELOG.md && git commit ../src/__fixtures__/package.json ../src/__fixtures__/CHANGELOG.md -m \\"chore(release): undefined\\""]</execSync>",
  },
  {
    "getGit": [
      {
        "commit": "<execSync>["git log <execSync>[\\"git..HEAD --pretty=oneline"]</execSync>",
        "isBreaking": false,
      },
    ],
  },
  {
    "getLinkUrls": {
      "baseUrl": undefined,
      "commitUrl": undefined,
      "compareUrl": undefined,
      "prUrl": undefined,
    },
  },
  {
    "getOverrideVersion": {
      "clean": "<semverClean>[null]</semverClean>",
      "version": undefined,
    },
  },
  {
    "getReleaseCommit": "<execSync>["git log HEAD --grep=\\"chore(release)\\" --pretty=oneline -1"]</execSync>",
  },
  {
    "getVersion": {
      "clean": "<semverClean>["<semverInc>[\\"0.0.100\\",\\"\\"]</semverInc>"]</semverClean>",
      "version": "<semverInc>["0.0.100",""]</semverInc>",
    },
  },
  {
    "runCmd": "<execSync>[null]</execSync>",
  },
]
`;

exports[`Commands should handle multiple formats for releaseTypeScope: releaseTypeScope, multi-list item 1`] = `
{
  "commitFiles": "<execSync>["git add ../src/__fixtures__/package.json ../src/__fixtures__/CHANGELOG.md && git commit ../src/__fixtures__/package.json ../src/__fixtures__/CHANGELOG.md -m \\"chore(release): undefined\\""]</execSync>",
  "getReleaseCommit": "<execSync>["git log HEAD --grep=\\"dolor sit\\" --pretty=oneline -1"]</execSync>",
}
`;

exports[`Commands should handle multiple formats for releaseTypeScope: releaseTypeScope, single list item 1`] = `
{
  "commitFiles": "<execSync>["git add ../src/__fixtures__/package.json ../src/__fixtures__/CHANGELOG.md && git commit ../src/__fixtures__/package.json ../src/__fixtures__/CHANGELOG.md -m \\"chore(release): undefined\\""]</execSync>",
  "getReleaseCommit": "<execSync>["git log HEAD --grep=\\"chore(release)\\" --pretty=oneline -1"]</execSync>",
}
`;

exports[`Commands should handle multiple formats for releaseTypeScope: releaseTypeScope, string 1`] = `
{
  "commitFiles": "<execSync>["git add ../src/__fixtures__/package.json ../src/__fixtures__/CHANGELOG.md && git commit ../src/__fixtures__/package.json ../src/__fixtures__/CHANGELOG.md -m \\"chore(release): undefined\\""]</execSync>",
  "getReleaseCommit": "<execSync>["git log HEAD --grep=\\"chore(release)\\" --pretty=oneline -1"]</execSync>",
}
`;
